test_e2e.sh

```diff
@@ -666,11 +666,56 @@ assert_file_eq "$ROOT"/link/file "$TESTCASE 1"
 # Wrap up
 pass
 
+##############################################
+# Test askpass_url
+##############################################
+testcase "askpass_url"
+echo "$TESTCASE 1" > "$REPO"/file
+NCPORT=$(($RANDOM+2000))
```

Comment:
Sorry, this is not what I meant by random.  This has a higher probability of success, but could randomly fail e2e tests.

I don't now if there's a better CLI way to bind to a kernel-assigned (never in conflict) port number, or if you need to run a dedicated container image..

Reply:
Looks like it's feasible to find a free port via bash.